pipeline/lib/pipeline-stack.ts

```diff
@@ -24,17 +24,19 @@ export class PipelineStack extends cdk.Stack {
       encryption: BucketEncryption.S3_MANAGED
     });
 
-    // Source (https://docs.aws.amazon.com/cdk/api/latest/docs/aws-codepipeline-actions-readme.html)
+    // Source
+    const gitRepo = "GIT_REPO_NAME"
+    const gitOwner = "GIT_REPO_OWNER"
+    const gitBranch = "GIT_BRANCH"
     const sourceArtifacts = new codepipeline.Artifact();
-    const sourceAction: GitHubSourceAction = new codepipeline_actions.GitHubSourceAction({
+    const sourceAction = new codepipeline_actions.CodeStarConnectionsSourceAction({
       actionName: 'Source',
-      owner: ssm.StringParameter.fromStringParameterName(this, 'GithubUsername', 'github_username').stringValue,
-      repo: 'aws-serverless-books-api-sample',
-      oauthToken: cdk.SecretValue.secretsManager('github_token', {jsonField: 'github_token'}),
-      output: sourceArtifacts,
-      branch: 'main',
-      trigger: codepipeline_actions.GitHubTrigger.WEBHOOK,
-      variablesNamespace: 'SourceVariables'
+      owner: gitOwner,
+      repo: gitRepo,
+      output: sourceArtifacts, 
+      branch: gitBranch,
+      connectionArn: 'arn:aws:codestar-connections:ap-southeast-1:585512738561:connection/d4127f97-aab8-4ac0-b95a-b7cf2bdc2d23',
```

Comment:
Looks like the connection ARN is specific for a given account and region. Would it be better to parameterize as we do for other input params such as gitXXX? Account Id and region can be obtained from the stack properties (see https://docs.aws.amazon.com/cdk/api/latest/docs/@aws-cdk_core.Stack.html)

Reply:
I can fix that by adding parameters like account and region, which is available in the stack, so we just need to provide the connection id. What do you think? Ignore that change and I will send another pull request.

Reply:
@jbernalvallejo 
I have submitted a commit with the changes and I believe that it also reflects this PR. Let me know if you can see it.

Reply:
I could use the region from the environment variables, however, codestar re uses the connection from other regions. In my case, I created my connection in ap-southeast-1 and I could re-use that in ap-southeast-2. So, I decided to add an extra variable for that.